Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-23.ca-central-1.compute.internal
Sha: b591bf1929ba57b5b806d5f40aa529593116f878
Number of Error(s): 9
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-14591/HPCC-Platform/roxie/ccd/ccdserver.cpp:1472:21: error: exception of type \xe2\x80\x98IException*\xe2\x80\x99 will be caught [-Werror]
/home/centos/smoketest/PR-14591/HPCC-Platform/roxie/ccd/ccdserver.cpp:1466:21: error:    by earlier handler for \xe2\x80\x98IException*\xe2\x80\x99 [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [roxie/ccd/CMakeFiles/ccd.dir/build.make:115: roxie/ccd/CMakeFiles/ccd.dir/ccdserver.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16255: roxie/ccd/CMakeFiles/ccd.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-02-17_13-19-28
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   466 sec (00:07:46)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   474 sec (00:07:54)
time logs end

[Build error:](https://gist.github.com/df41fdd41b849c8ab1c7242e540779f6)
Build log [Build log](https://gist.github.com/HPCCSmoketest/df41fdd41b849c8ab1c7242e540779f6#file-result-21-02-17-13-11-22-log)

--- a/roxie/ccd/ccdserver.cpp
+++ b/roxie/ccd/ccdserver.cpp
@@ -1459,7 +1459,21 @@ public:
                         throw makeStringExceptionV(ROXIE_INTERNAL_ERROR, "STATE: activity %d reset without stop", activityId);
                     if (traceStartStop || traceLevel > 2)
                         CTXLOG("STATE: activity %d reset without stop", activityId);
-                    stop();
+                    try
+                    {
+                        stop();
+                    }
+                    catch (IException *E)
+                    {
+                        EXCLOG(E, "Unexpected exception in stop() called from reset()");
+                        printStackReport();
+                        ::Release(E);
+                    }
+                    catch (...)
+                    {
+                        DBGLOG("Unexpected unknown exception in stop() called from reset()");
+                        printStackReport();
+                    }
                 }
                 state = STATEreset;
 #ifdef TRACE_STARTSTOP

